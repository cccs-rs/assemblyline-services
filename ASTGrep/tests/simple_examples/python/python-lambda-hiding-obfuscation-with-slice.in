func1 = lambda a: sth(a)
func2 = lambda a: sth(a[1:4])
func3 = lambda a: module.sth(a)

func1("xxx")
func2("123456789")
func3(16)

_ = lambda __ : some_func("aaa").attribute("xy").final(__[:1])
_("bbb")

x = (lambda t: y(t))
x(123)

func4, func5 = lambda b: sth4(b), (lambda b: sth5(b))
func4("xxx4")
func5("xxx5")

func6 = lambda: sth6
func6()("xxx6")
func6("xxx6")

func7 = lambda a: a(sth)
func7(xxx7)

func8 = lambda a: a[key]
func8(xxx8)

func9 = lambda(a, b): a(b)
func10 = lambda(b, a): a(b)

func9(x9, y9)
func10(x10, y10)

func11 = lambda a, b: a(b)
func11(x11, y11)
