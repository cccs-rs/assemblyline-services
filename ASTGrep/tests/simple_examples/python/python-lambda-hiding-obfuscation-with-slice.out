func1 = sth
func2 = lambda a: sth(a[1:4])
func3 = module.sth

sth("xxx")
sth('234')
module.sth(16)

_ = lambda __ : some_func("aaa").attribute("xy").final(__[:1])
some_func("aaa").attribute("xy").final('b')

x = (y)
y(123)

func4, func5 = sth4, (sth5)
sth4("xxx4")
sth5("xxx5")

func6 = lambda: sth6
sth6("xxx6")
func6("xxx6")

func7 = lambda a: a(sth)
xxx7(sth)

func8 = lambda a: a[key]
xxx8[key]

func9 = lambda(a, b): a(b)
func10 = lambda(b, a): a(b)

x9(y9)
y10(x10)

func11 = lambda a, b: a(b)
x11(y11)
